lib/service/translation/amazon_translate.dart

```diff
@@ -0,0 +1,314 @@
+    );
+
+    if (!apiResult.succeeded) {
+      logger.warning('Translation failed');
+      return source;
+    }
+
+    return apiResult.value!.translations.first.translatedText;
+  }
+
+  /// Translates given texts to specified language
+  /// [sources] - list of text which should be translated
+  /// [sourceLanguage] - the language in which [sources] were given
+  /// [target] - language to which [sources] should be translated
+  @override
+  Future<List<String>> translateBulkToSingleTarget(
+    List<String> sources,
+    LanguageCode sourceLanguage,
+    LanguageCode target,
+  ) async {
+    logger.info('Translate ${sources.length} texts from $sourceLanguage to $target');
+
+    final apiResult = await client.translate(
+      logger: logger,
+      messages: sources,
+      sourceLanguageCode: sourceLanguage,
+      targetLanguageCode: target,
+    );
+
+    if (!apiResult.succeeded) {
+      logger.warning('Translation failed');
+      return sources;
+    }
+
+    return apiResult.value!.translations.map((e) => e.translatedText).toList();
+  }
+}
+
+/// Client for Amazon Translate API
+/// Generates signed request and sends it to Amazon Translate API
+/// [accessKeyId] - AWS Access Key Id, read mode here:
+/// https://docs.aws.amazon.com/IAM/latest/UserGuide/security-creds.html
+/// [secretAccessKey] - AWS Secret Access Key, read mode here:
+/// https://docs.aws.amazon.com/IAM/latest/UserGuide/security-creds.html
+/// [region] - AWS Service Endpoint Region, read more here:
+/// https://docs.aws.amazon.com/general/latest/gr/rande.html
+/// [service] - AWS Service Name, in this case it is "translate"
+class AmazonTranslateClient {
+  final String accessKeyId;
+  final String secretAccessKey;
+  final String region;
+  final String service;
+
+  AmazonTranslateClient._({
+    required this.accessKeyId,
+    required this.secretAccessKey,
+    required this.region,
+    required this.service,
+  });
+
+  factory AmazonTranslateClient.translate({
+    required String accessKeyId,
+    required String secretAccessKey,
+    required String region,
+  }) =>
+      AmazonTranslateClient._(
+        accessKeyId: accessKeyId,
+        secretAccessKey: secretAccessKey,
+        region: region,
+        service: "translate",
+      );
+
+  String get host => "$service.$region.amazonaws.com";
+
+  String get endpoint => 'https://' + host;
+
+  String get target => 'AWSShineFrontendService_20170701.TranslateText';
+
+  String get credentialScope => "$date/$region/$service/aws4_request";
+
+  String get contentType => 'application/x-amz-json-1.1; charset=utf-8';
+
+  String get signedHeaders => 'content-type;host;x-amz-date;x-amz-target';
+
+  String get amzDateTime => DateFormat("yyyyMMddTHHmm00'Z'").format(DateTime.now().toUtc());
+
+  String get date => DateFormat("yyyyMMdd").format(DateTime.now().toUtc());
+
+  String get algorithm => 'AWS4-HMAC-SHA256';
+
+  HttpClient buildClient(Logger logger, String message) {
```

Comment:
Please update visibility of methods in this class
It looks like the only public method is `translate()`. If so, please modify other methods to be private, for example `buildClient()` -> `_buildClient()`, etc